packages/vscode-extension/src/debug/teamsfxDebugProvider.ts

```diff
@@ -9,20 +10,39 @@ import * as commonUtils from "./commonUtils";
 export interface TeamsfxDebugConfiguration extends vscode.DebugConfiguration {
   teamsfxEnv?: string;
   teamsfxAppId?: string;
+  teamsfxCorrelationId?: string;
 }
 
 export class TeamsfxDebugProvider implements vscode.DebugConfigurationProvider {
   public async resolveDebugConfiguration?(
     folder: vscode.WorkspaceFolder | undefined,
     debugConfiguration: TeamsfxDebugConfiguration,
     token?: vscode.CancellationToken
+  ): Promise<vscode.DebugConfiguration | undefined> {
+    return Correlator.runWithId(
```

Comment:
Will local debug correlator nest with command correlator?

Reply:
No. Because Correlation is related to a callstack. As long as the two function call are not in the same call stack, they will not overlap. 

- TeamsfxDebugProvider is called by vscode when user clicks Debug or hits F5
- Other `Correlator.run()` are called in command handlers and task providers, so they will not overlap

Even if they overlap AsyncLocalStorage can handle it correctly. With a quick test:
```
> cat asyncstore.js                                                                                                                                                                                                             16:44:55
const { AsyncLocalStorage } = require("async_hooks");
const l = new AsyncLocalStorage();

l.run("1", () => {
  console.log("before " + l.getStore());
  l.run("2", () => {
    console.log("inner " + l.getStore());
  });
  console.log("after " + l.getStore());
})
[16:44:58] alexwang@alex-p330 ~  
> node asyncstore.js                                                                                                                                                                                                            16:44:58
before 1
inner 2
after 1   <======= this is 1, not 2. so they handle re-entrance correctly
```